internal/util/string.go

```diff
@@ -18,3 +18,11 @@ func BaseString(obj any) string {
 func JoinStrings(delim string, parts ...string) string {
 	return strings.Join(parts, delim)
 }
+
+func GetFirstNonEmptyString(primary string, fallback string) string {
```

Comment:
This name is a bit off.
What if both are empty?

This concept is often called a "default",
so maybe "GetStringWithDefault()"?